src/std/ucs2.c

```diff
@@ -83,45 +67,47 @@ uchar *ustrdup( const uchar *str ) {
 	return d;
 }
 
-double utod( const uchar *str, uchar **end ) {
+double utod( const uchar *str, const uchar **end ) {
 	char buf[31];
 	char *bend;
+	double result;
 	int i = 0;
-	double v;
-	while( is_space_char(*str) ) str++;
 	while( i < 30 ) {
-		int c = *str++;
+		int c = str[i];
 		if( (c < '0' || c > '9') && c != '.' && c != 'e' && c != 'E' && c != '-' && c != '+' )
 			break;
 		buf[i++] = (char)c;
 	}
 	buf[i] = 0;
-	v = strtod(buf,&bend);
-	*end = (uchar*)(str - 1) + (bend - buf);
-	return v;
+	result = strtod(buf,&bend);
+	*end = str + (bend - buf);
+	return result;
 }
 
-int utoi( const uchar *str, uchar **end ) {
+int utoi( const uchar *str, const uchar **end ) {
 	char buf[17];
 	char *bend;
+	int result;
 	int i = 0;
-	int v;
-	while( is_space_char(*str) ) str++;
+	uchar sign = str[0];
```

Comment:
Shouldn't this account for an empty string, in which case `str[0]` is out-of-bounds?

Reply:
Strings are null terminated, so an empty string would be `"\0"`, in that case `str[0]` would be `'\0'`. It's the same as `*str` which was done before.

Reply:
Oh right, and I feel like you told me that before, haha...